includes/discovery/functions.inc.php

```diff
@@ -76,15 +76,17 @@ function discover_new_device($hostname, $device = [], $method = '', $interface =
     }
 
     try {
-        $remote_device_id = addHost($hostname, '', '161', 'udp', Config::get('default_poller_group'));
+        $remote_device_id = addHost($hostname, '', '161', 'udp', $device['poller_group']); // discover with actual poller group
         $remote_device = device_by_id_cache($remote_device_id, 1);
         echo '+[' . $remote_device['hostname'] . '(' . $remote_device['device_id'] . ')]';
         discover_device($remote_device);
         device_by_id_cache($remote_device_id, 1);
 
         if ($remote_device_id && is_array($device) && ! empty($method)) {
             $extra_log = '';
-            $int = cleanPort($interface);
+            // cleanPort expect array
+            $ifArray = ['ifName' => $interface, 'ifDescr' => $interface, 'ifAlias' => $interface, 'device_id' => $device['device_id']];
+            $int = cleanPort($ifArray);
```

Comment:
I think the correct fix is to move `$int = cleanPort($interface);` down one line inside the `is_array($int)` block

Reply:
(and change $int to $interface in that check)

Reply:
> I think the correct fix is to move `$int = cleanPort($interface);` down one line inside the `is_array($int)` block

Hi @murrant 
i like to ask you to take careful look at discovery-protocols.inc.php file
you will see that every time function is called similar to this:
```$remote_device_id = discover_new_device($lldp['lldpRemSysName'], $device, 'LLDP', $interface);```

so, it always pass plain variable

if i put ```is_array($interface)``` check in discover_new_device function, the EXTRA_LOG will bi never triggered, because we always send flat $interface variable like 'gigabitethernet 1/0/11'

my intention was to convert plain $interface to array so we could get this extra_log line in eventlog

but if you insist ...

Reply:
> (and change $int to $interface in that check)

Ok, done that way as you suggested

Reply:
@Npeca75 if you scroll up a bit in the code, you see `$interface = get_port_by_id($local_port_id);` get_port_by_id() returns an array, not a string.